Habitica/src/main/java/com/habitrpg/android/habitica/receivers/TodoReceiver.java

```diff
@@ -0,0 +1,56 @@
+package com.habitrpg.android.habitica.receivers;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.habitrpg.android.habitica.R;
+import com.habitrpg.android.habitica.ui.activities.MainActivity;
+import com.habitrpg.android.habitica.ui.helpers.TaskAlarmManager;
+
+/**
+ * Created by keithholliday on 5/29/16.
+ */
+public class TodoReceiver  extends BroadcastReceiver {
+
+    private Context context;
+
+    private String taskId;
+    private String taskTitle;
+    private TaskAlarmManager taskAlarmManager;
+
+    @Override
+    public void onReceive(Context arg0, Intent arg1) {
+        context = arg0;
+        taskAlarmManager = TaskAlarmManager.getInstance(context);
+
+        Bundle extras = arg1.getExtras();
+        if (extras != null) {
+            taskTitle = extras.getString(TaskAlarmManager.TASK_NAME_INTENT_KEY);
+            taskId = extras.getString(TaskAlarmManager.TASK_ID_INTENT_KEY);
+        }
+
+        createNotification();
+    }
+
+    public void createNotification() {
+        Intent intent = new Intent(context, MainActivity.class);
+        PendingIntent pIntent = PendingIntent.getActivity(context, (int) System.currentTimeMillis(), intent, 0);
+
+        Notification noti = new Notification.Builder(context)
+                .setContentTitle("Your todo " + taskTitle + " is due!")
```

Comment:
On iOS we just use the task title for the notification. I would also do that here. Alternatively, I think the taskTitle should be styled differently / surrounded by " ", so that it's clear what the task part is.